File: clothing_backend/models/voucher_model.js
import { DataTypes } from 'sequelize';
import sequelize from '../connection/mysql.js';
import { v4 as uuidv4 } from 'uuid';
import User from './user_model.js';

const Voucher = sequelize.define('voucher', {
    id: {
        type: DataTypes.STRING,
        primaryKey: true,
        defaultValue: () => uuidv4(),
    },
    title : {
        type: DataTypes.STRING,
        allowNull: false,
    },
    description :{
        type: DataTypes.STRING,
        allowNull: false,
    },
    discount:{
        type: DataTypes.FLOAT,
        allowNull: false,
    },
    type:{
        type: DataTypes.STRING,
        allowNull: false,
    },
    start_at:{
        type: DataTypes.DATE,
        allowNull: false,
    },
    end_at:{
        type: DataTypes.DATE,
        allowNull: false,
    },
    is_public:{
        type: DataTypes.BOOLEAN,
        allowNull: false,
    },

}, {
    timestamps: true //Tự động thêm các trường createdAt và updatedAt
});


const VoucherUser = sequelize.define("user_has_voucher",{
    id: {
        type: DataTypes.STRING,
        primaryKey: true,
        defaultValue: () => uuidv4(),
    },
    quantity : {
        type: DataTypes.INTEGER,
        allowNull: false,
    },
    voucher_id:{
        type: DataTypes.STRING,
        allowNull: false, 
    },
    user_id :{
        type: DataTypes.STRING,
        allowNull: true,
    },
    condition:{
        type: DataTypes.STRING,
        allowNull: false, 
    },
})


export default {Voucher,VoucherUser};

File: clothing-store-client/app/src/main/java/vn/clothing/store/models/VoucherModel.kt
package vn.clothing.store.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.Date

class VoucherModel : Serializable{
    var id:String?=null
    var title:String?=null
    var description:String?=null
    var discount:Double?=null
    var type:String?=null
    @SerializedName("start_at")
    var startAt:Date?=null
    @SerializedName("end_at")
    var endAt:Date?=null
    var quantity:Int = 0
    var used:Int = 0
    @SerializedName("user_id")
    var userId:String?=null
}

